### PR TITLE
Modify device model to match the 0.7 spec - (Refer to new PR due to more changes to spec)

### PR DESCRIPTION
**Pull request by ankur325**
This PR contains the following changes

- Device Model schema changes to match what is in 0.7 version of the spec
- CLI changes for the new Device Model (updated the flags accordingly)
- Updates to Transaction and How To Guides
- Unit and Integration tests for the new changes

### CODE REVIEW

File: x/modelinfo/genesis.go [hunk 1]
```diff
@@ -53,24 +53,25 @@ func ValidateGenesis(data GenesisState) error {
 			return sdk.ErrUnknownRequest(fmt.Sprintf("Invalid ModelInfo: Missed SKU. Value: %v", record))
 		}
 
-		if record.HardwareVersion == "" {
+		if record.HardwareVersion == 0 {
 			return sdk.ErrUnknownRequest(fmt.Sprintf("Invalid ModelInfo: Missed HardwareVersion. Value: %v", record))
 		}
+		// TODO we should reuse validation
```
Comment (ashcherbakov) [inline x/modelinfo/genesis.go:59]: I believe we should have validation for other fields here (not only HardwareVersion), shouldn't we?

File: x/modelinfo/handler_test.go [hunk 2]
```diff
@@ -40,8 +40,10 @@ func TestHandler_AddModel(t *testing.T) {
 	receivedModelInfo := queryModelInfo(setup, modelInfo.VID, modelInfo.PID)
 
 	// check
+	// TODO do a smart check
+
 	require.Equal(t, receivedModelInfo.VID, modelInfo.VID)
-	require.Equal(t, receivedModelInfo.PID, modelInfo.PID)
+	/*require.Equal(t, receivedModelInfo.PID, modelInfo.PID)
```
Comment (ashcherbakov) [inline x/modelinfo/handler_test.go:46]: Should we uncomment these lines?

File: x/modelinfo/handler_test.go [hunk 3]
```diff
@@ -81,18 +83,7 @@ func TestHandler_UpdateModel(t *testing.T) {
 	require.Equal(t, receivedModelInfo.VID, msgAddModelInfo.VID)
 	require.Equal(t, receivedModelInfo.PID, msgAddModelInfo.PID)
 	require.Equal(t, receivedModelInfo.CID, msgUpdateModelInfo.CID)
-	require.Equal(t, receivedModelInfo.Version, msgAddModelInfo.Version)
```
Comment (ashcherbakov) [inline x/modelinfo/handler_test.go:84]: Why these lines have been removed?

File: docs/cli-help.md [hunk 4]
```diff
@@ -394,48 +394,74 @@ The set of commands that allows you to manage model infos.
   Role: `Vendor`
   
   Command: `dclcli tx modelinfo add-model --vid=<uint16> --pid=<uint16> --name=<string> --description=<string or path> --sku=<string> 
---firmware-version=<string> --hardware-version=<string> --tis-or-trp-testing-completed=<bool> --from=<account>`
+--software-version=<uint32> --software-version-string=<string> --hardware-version=<uint32> --hardware-version-string=<string> --cd-version-number=<uint16> 
+--from=<account>`
 
   Flags:
   - vid: `uint16` -  model vendor ID (positive non-zero)
```
[the remark this answers is not in the document]
Comment (ankur325) [reply docs/cli-help.md:401]: Yes, I think camelCase makes it easier, even while typing. Will make that change and update this PR.

File: docs/cli-help.md [hunk 5]
```diff
@@ -394,48 +394,74 @@ The set of commands that allows you to manage model infos.
...
   - pid: `uint16` -  model product ID (positive non-zero)
   - name: `string` -  model name
```
[the remark this answers is not in the document]
Comment (ankur325) [reply docs/cli-help.md:403]: I tried to reuse existing field name, but I agree this can be renamed to match the spec

File: go.mod [hunk 6]
```diff
@@ -14,4 +17,5 @@ require (
 	github.com/tendermint/go-amino v0.15.1
 	github.com/tendermint/tendermint v0.32.8
 	github.com/tendermint/tm-db v0.2.0
+	golang.org/x/sys v0.0.0-20210616094352-59db8d763f22 // indirect
```
Comment (ashcherbakov) [inline go.mod:20]: Why do we need this dependency?

File: utils/conversions/conversions.go [hunk 7]
```diff
@@ -21,6 +21,15 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
+func ParseUInt8FromString(str string) (uint8, sdk.Error) {
+	val, err := strconv.ParseUint(str, 10, 8)
+	if err != nil {
+		return 0, sdk.ErrUnknownRequest(fmt.Sprintf("Parsing Error: \"%v\" must be 16 bit unsigned integer", str))
```
[the remark this answers is not in the document]
Comment (ashcherbakov) [reply utils/conversions/conversions.go:27]: Should we apply that fix?

File: utils/conversions/conversions.go [hunk 8]
```diff
@@ -30,6 +39,15 @@ func ParseUInt16FromString(str string) (uint16, sdk.Error) {
 	return uint16(val), nil
 }
 
+func ParseUInt32FromString(str string) (uint32, sdk.Error) {
+	val, err := strconv.ParseUint(str, 10, 32)
+	if err != nil {
+		return 0, sdk.ErrUnknownRequest(fmt.Sprintf("Parsing Error: \"%v\" must be 16 bit unsigned integer", str))
```
[the remark this answers is not in the document]
Comment (ashcherbakov) [reply utils/conversions/conversions.go:45]: Should we apply this fix?

File: x/modelinfo/client/rest/tx.go [hunk 9]
```diff
@@ -23,35 +23,14 @@ import (
...
 type UpdateModelInfoRequest struct {
-	BaseReq                  restTypes.BaseReq `json:"base_req"`
-	VID                      uint16            `json:"vid"`
-	PID                      uint16            `json:"pid"`
-	CID                      uint16            `json:"cid,omitempty"`
-	Description              string            `json:"description,omitempty"`
-	OtaURL                   string            `json:"ota_url,omitempty"`
-	Custom                   string            `json:"custom,omitempty"`
-	TisOrTrpTestingCompleted bool              `json:"tis_or_trp_testing_completed"`
+	Model   types.Model       `json:"model"`
```
Comment (ashcherbakov) [inline x/modelinfo/client/rest/tx.go:32]: I believe we don't allow changing of any field in Edit command. So, should we use another type here, not `types.Model` which has all the model fields?

File: x/modelinfo/handler_test.go [hunk 10]
```diff
@@ -40,21 +40,9 @@ func TestHandler_AddModel(t *testing.T) {
...
-	require.Equal(t, receivedModelInfo.SKU, modelInfo.SKU)
-	require.Equal(t, receivedModelInfo.HardwareVersion, modelInfo.HardwareVersion)
-	require.Equal(t, receivedModelInfo.FirmwareVersion, modelInfo.FirmwareVersion)
-	require.Equal(t, receivedModelInfo.OtaURL, modelInfo.OtaURL)
-	require.Equal(t, receivedModelInfo.OtaChecksum, modelInfo.OtaChecksum)
-	require.Equal(t, receivedModelInfo.OtaChecksumType, modelInfo.OtaChecksumType)
-	require.Equal(t, receivedModelInfo.Custom, modelInfo.Custom)
-	require.Equal(t, receivedModelInfo.TisOrTrpTestingCompleted, modelInfo.TisOrTrpTestingCompleted)
-	require.Equal(t, receivedModelInfo.Owner, modelInfo.Signer)
+	require.Equal(t, receivedModelInfo.Model.VID, modelInfo.VID)
```
Comment (ashcherbakov) [inline x/modelinfo/handler_test.go:43]: Why do we check for VID, PID, CID here? Can we check for all the model fields?

File: integration_tests/cli/modelinfo-demo.sh [hunk 11]
```diff
@@ -25,17 +25,17 @@ create_new_account vendor_account "Vendor"
 
 vid=$RANDOM
 pid=$RANDOM
-name="Device #1"
+productName="Device #1"
 echo "Add Model with VID: $vid PID: $pid"
-result=$(echo "test1234" | dclcli tx modelinfo add-model --vid=$vid --pid=$pid --name="$name" --description="Device Description" --sku="SKU12FS" --hardware-version="1.1" --firmware-version="2.0" --tis-or-trp-testing-completed=false --from $vendor_account --yes)
+result=$(echo "test1234" | dclcli tx modelinfo add-model --vid=$vid --pid=$pid --supportURL="https://originalsupporturl.test" --productName="$productName" --description="Device Description" --sku="SKU12FS" --softwareVersion="10123" --softwareVersionString="1.0b123"  --hardwareVersion="5123" --hardwareVersionString="5.1.23"  --cdVersionNumber="32" --from $vendor_account --yes)
```
Comment (ashcherbakov) [inline integration_tests/cli/modelinfo-demo.sh:30]: Can we have an integration test where we create a model with all possible fields ?

**Pull request closed without merging**